addons/hr_contract/models/resource_resource.py

```diff
@@ -54,3 +54,32 @@ def _get_calendars_validity_within_period(self, start, end, default_company=None
                 self.env['resource.calendar.attendance']
             )])
         return calendars_within_period_per_resource
+
+    def _is_flexible_at(self, date_start, date_end, tz):
+        flexible_resources = defaultdict(lambda: self.env['resource.resource'])
+        valid_contracts = self.env['hr.contract'].sudo().search([
+            ('employee_id', 'in', self.employee_id.ids),
+            ('active', '=', True),
+        ]).grouped('employee_id')
```

Comment:
By default, an active test is done on searches so no need to make it explicit 🙂 
You can also restrict a bit more
Last point, `_read_group` is slightly more efficient than group on a search (though I agree it's less readable)
```suggestion
        valid_contracts = dict(self.env['hr.contract'].sudo()._read_group(
            domain=[
                ('employee_id', 'in', self.employee_id.ids),
            ],
            groupby=['employee_id'],
            aggregates=['id:recordset'],
        ))
```